.github/workflows/main.yaml

```diff
@@ -41,16 +41,47 @@ on:
         type: string
         default: "latest"
       refresh:
+
         required: false
         type: string
         default: "0"
 permissions:
   contents: read
 
 jobs:
+  get_versions:
+    runs-on: ubuntu-latest
+    name: Setup the correct version of python and github os for pyodide version
+    outputs:
+      os: ${{ steps.fix_runner_os.outputs.os }}
+      python: ${{ steps.get_python_version.outputs.python }}
+      pythonexec: ${{ steps.get_python_version.outputs.pythonexec }}
+    steps:
+      - name: Get python version for a particular pyodide version
+        id: get_python_version
+        run: |
+          case "${{inputs.pyodide-version}}" in
+            *)
+              echo "python=3.10.2" >> $GITHUB_OUTPUT
+              echo "pythonexec=python310" >> $GITHUB_OUTPUT
+              echo "Using python 3.10.2 for pyodide ${{inputs.pyodide-version}}"
+              ;;
+          esac
+      - name: Fix runner os
+        id: fix_runner_os
+        run: |
+          if [ "${{ steps.get_python_version.outputs.python }}" = "3.10.2"] && [ "${{ inputs.os}}" = "ubuntu-latest" ]
```

Comment:
Could you also add `ubuntu-22.04`?

Reply:
Done these. 

The auto-python version choosing is already in this code - if you add a case to the case statement in main.yaml at line 58, you can set different python versions based on pyodide versions.